### ISSUE TITLE
Easy way to corrupt data

### ISSUE THREAD

**Issue by doushe821** [Dariazeml1007/VulnerableStack#2]
![image](https://github.com/user-attachments/assets/eede2d45-49df-4454-a31a-d86bb4af46a4)

Also it seems like there is no proper size check in stackpop.

**Comment by Dariazeml1007** [comment]
Yeah, you have found all mistakes (easy to corrupt structure and wrong size check in stackpop)
